src/atomate2/common/schemas/pheasy.py

```diff
@@ -227,34 +229,36 @@ def from_forces_born(
             is_symmetry=sym_reduce,
         )
 
-        # Start from here, we use the pheasy code to extract the force constants.
+        # Write the POSCAR and SPOSCAR files for the input of pheasy code
         supercell = phonon.get_supercell()
         write_vasp("POSCAR", cell)
         write_vasp("SPOSCAR", supercell)
 
         # get the force-displacement dataset from previous calculations
-        set_of_forces = [np.array(forces) for forces in displacement_data["forces"]]
-        set_of_forces_a_o = np.array(set_of_forces)
+        dataset_forces = [np.array(forces) for forces in displacement_data["forces"]]
+        dataset_forces_array = np.array(dataset_forces)
+
+        # To deduct the residual forces on an equilibrium structure to eliminate the fitting error
+        dataset_forces_array_rr = dataset_forces_array - dataset_forces_array[-1, :, :]
 
-        # deduct the residual forces on the equilibrium structure
-        set_of_forces_a_t = set_of_forces_a_o - set_of_forces_a_o[-1, :, :]
-        set_of_forces_a = set_of_forces_a_t[:-1, :, :]
-        set_of_disps = [np.array(disps.cart_coords) for disps in displacement_data["displaced_structures"]]
+        # force matrix on the displaced structures
+        dataset_forces_array_disp = dataset_forces_array_rr[:-1, :, :]
+        dataset_disps = [np.array(disps.cart_coords) for disps in displacement_data["displaced_structures"]]
 
         # get the displacement dataset
-        set_of_disps_m_o = np.round((set_of_disps - supercell.get_positions()), 
+        dataset_disps_array_rr = np.round((dataset_disps - supercell.get_positions()), 
                                     decimals=16).astype('double')
-        set_of_disps_m = set_of_disps_m_o[:-1, :, :]
+        dataset_disps_array_use = dataset_disps_array_rr[:-1, :, :]
 
-        # get the number of displacements
-        num_har = set_of_disps_m.shape[0]
+        # get the number of displacements for harmonic phonon calculation
+        num_har = dataset_disps_array_use.shape[0]
 
         # save the displacement and force matrix in the current directory
         # for the future use by pheasy code 
         with open("disp_matrix.pkl","wb") as file: 
-             pickle.dump(set_of_disps_m,file)
+             pickle.dump(dataset_disps_array_use,file)
         with open("force_matrix.pkl","wb") as file:
-             pickle.dump(set_of_forces_a,file)
+             pickle.dump(dataset_forces_array_disp,file)
```

Comment:
Can you only use pickle here?

Reply:
other format also should be fine, any suggestions? because in pheasy, it uses the pickle, so i just simply follow it.

Reply:
> Can you only use pickle here?

other format also should be fine, any suggestions? because in pheasy, it uses the pickle, so i just simply follow it.